tools/shared/src/main/scala/org/scalajs/core/tools/linker/backend/emitter/Emitter.scala

```diff
@@ -198,16 +198,32 @@ final class Emitter private (semantics: Semantics, outputMode: OutputMode,
         }
 
       case ModuleKind.ESModule =>
-        foreachImportedModule { (module, pos0) =>
+        foreachImportedModule { (module, pos0, path) =>
           implicit val pos = pos0
           val from = js.StringLiteral(module)
           val moduleBinding = jsGen.envModuleField(module).ident
-          val importStat = js.ImportNamespace(moduleBinding, from)
-          builder.addJSTree(importStat)
+          path match {
+            // namespace import or default import
+            case Nil | List("default") =>
+              val importStat = js.ImportNamespace(moduleBinding, from)
+              builder.addJSTree(importStat)
+            case _ =>
+              val bindings = path.map(p => (js.ExportName(p), js.Ident(p)))
```

Comment:
I think you may be misunderstanding what a `path` is. In the following import:
```scala
@js.native
@JSImport("foo", "Bar.Foobar")
```
the `path` will be `List("Bar", "Foobar")`.

Only `path.head` is relevant to generate named imports. further selections in the `path` must remain as deeper selections, and should not participate in the `import` statement.